tests/pds/dataservice_test.go

```diff
@@ -745,6 +745,86 @@ var _ = Describe("{UpgradeDataServiceImage}", func() {
 	})
 })
 
+var _ = Describe("{DrainAndDecommissionNode}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("DrainAndDecommissionNode", "Deploys a data service, drains one selected node, decommissions that node", pdsLabels, 0)
+	})
+
+	It("Drain and Decommission a node", func() {
+		Step("Deploy, Validate, Drain Node, Cordon Node, Validate Data Service, Run Workload on Data Service, Uncordon Node", func() {
+			for _, ds := range params.DataServiceToTest {
+				isDeploymentsDeleted = false
+				deployment, _, _, err = DeployandValidateDataServices(ds, params.InfraToTest.Namespace, tenantID, projectID)
+				log.FailOnError(err, fmt.Sprintf("Error while deploying data services %s", ds.Name))
+				log.InfoD("Running Check Node DataService Func %v ", ds.Name)
+				nodes, err := pdslib.GetNodesOfSS(*deployment.ClusterResourceName, namespace)
+				log.FailOnError(err, "Cannot fetch nodes of the running Data Service")
+				var nodeName string
+				nodeName = nodes[0].Name // Selecting the 1st node in the list to cordon
+				Step("Drain Pods from a node", func() {
+					podsList, err := pdslib.GetPodsOfSsByNode(*deployment.ClusterResourceName, nodeName, namespace)
+					log.FailOnError(err, fmt.Sprintf("Pod not found on this Node : %s", nodeName))
+					log.InfoD("Pods found on %v node. Trying to Drain pods from this node now.", nodeName)
+					err = k8sCore.DrainPodsFromNode(nodeName, podsList, timeOut, maxtimeInterval)
+					log.FailOnError(err, fmt.Sprintf("Draining pod from the node %s failed", nodeName))
+					log.InfoD("Pods successfully drained from the node %s", nodeName)
+				})
+				Step("Cordon Selected Node", func() {
+					err = k8sCore.CordonNode(nodeName, timeOut, maxtimeInterval)
```

Comment:
I guess no need to cordon off the node after draining, as the func k8sCore.DrainPodsFromNode cordon the node first and later delete all the required pods.
Also as part of the TestPostRun please add the uncordoning of the node, as the test might gonna fail in between, which in term leaves the node cordoned.

Reply:
Discussed this in the meeting